test/net/sourceforge/kolmafia/textui/command/MirrorCommandTest.java

```diff
@@ -0,0 +1,163 @@
+public class MirrorCommandTest extends AbstractCommandTestBase {
+
+  public MirrorCommandTest() {
+    this.command = "mirror";
+  }
+
+  @BeforeEach
+  public void stopMirrorLogging() {
+    execute("");
+  }
+
+  private String getMirrorLog(String fileName) {
+    StringBuilder builder = new StringBuilder();
+    try (BufferedReader reader =
+        FileUtilities.getReader(new File(KoLConstants.ROOT_LOCATION, "chats/" + fileName))) {
+      String line;
+      // Read the full file to a string
+      while ((line = FileUtilities.readLine(reader)) != null) {
+        if (builder.length() > 0) {
+          builder.append("\n");
+        }
+        builder.append(line);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    return builder.toString();
+  }
+
+  @Test
+  @Disabled("Using > does force colorization. ")
+  public void testMirrorWritesHTML() {
+    // Open the mirror
+    execute("chats/test_writes_html.txt");
+    // When `> ` is used as a prefix, the RequestLogger will colorize it assuming its a command
+    // input.
+    RequestLogger.printLine("> Fake command input");
+    // Log another line, this should be raw.
+    RequestLogger.printLine("Raw Line");
+    // Close the mirror
+    execute("");
+    String mirrorOutput = getMirrorLog("test_writes_html.txt");
+    assertTrue(mirrorOutput.contains("<font color=olive>> Fake command input</font><br>"));
+    assertTrue(mirrorOutput.contains("Raw Line<br>"));
+  }
+
+  @Test
+  public void testMirrorDoesNotAddHTML() {
+    // Open the mirror
+    execute("chats/test_writes.txt");
+    // Log a raw line, this should be raw.
+    RequestLogger.printLine("Raw Line");
+    // Close the mirror
+    execute("");
+    String mirrorOutput = getMirrorLog("test_writes.txt");
+    assertTrue(mirrorOutput.contains("Raw Line"));
+  }
+
+  @Test
+  public void testMirrorClose() {
+    execute("chats/test_mirror_closes");
+    // Write to the mirror log
+    RequestLogger.printLine("Some input");
+    // Close the mirror
+    execute("");
+    // Write to the command buffer again
+    RequestLogger.printLine("Should not be seen");
+    // Now read the log and verify that it was not written
+    String mirrorOutput = getMirrorLog("test_mirror_closes.txt");
+    // Test that the second line is not in the output
+    assertFalse(mirrorOutput.contains("Should not be seen"));
+  }
+
+  @Test
+  public void testMirrorListStrings() {
+    execute("chats/test_mirror_list_strings");
+    List<String> list = new ArrayList<>();
+    list.add("Line 1");
+    list.add("Line 2");
+    RequestLogger.printList(list);
+    // Close the mirror
+    execute("");
+    // Read the log
+    String mirrorOutput = getMirrorLog("test_mirror_list_strings.txt");
+    // Test that the output matches as expected
+    assertEquals("Line 1\n" + "Line 2", mirrorOutput);
+  }
+
+  @Test
+  public void testMirrorListSkills() {
+    try (var cleanups = new Cleanups(withSkill(1), withSkill(2))) {
```

Comment:
This doesn't work on all OS I think? @midgleyc couldn't you not do this

Reply:
couldn't I not do what?

This is fine, it just makes the IDE complain that `cleanups` is unused (which it is, but you need to assign to a variable in a try-with-resources), hence why I normally go for `var cleanups = ...; try (cleanups)`.

Reply:
IntelliJ isn't flagging var as unused for me.

Reply:
does it flag any variables as unused? If it does there might be a setting I'm missing.

Reply:
Yes.  I normally use Analyze->Inspect.  I introduced an unused variable and it was recognized as such (by being gray in the IDE text display) and flagged by the inspection.

Reply:
So I did a recompile from within the IDE and the complier complained about unused.  The manually run analysis did not.  So I am seeing what you did

Reply:
Ah it was an IntelliJ thing. So it sounds like you need to change this frono?